aegea/batch.py

```diff
@@ -27,26 +27,35 @@
 ebs_vol_mgr_shellcode = """iid=$(http http://169.254.169.254/latest/dynamic/instance-identity/document)
 aws configure set default.region $(echo "$iid" | jq -r .region)
 az=$(echo "$iid" | jq -r .availabilityZone)
+echo Creating volume >&2
 vid=$(aws ec2 create-volume --availability-zone $az --size %s --volume-type st1 | jq -r .VolumeId)
 aws ec2 create-tags --resource $vid --tags Key=aegea_batch_job,Value=$AWS_BATCH_JOB_ID
-trap "umount /mnt || umount -l /mnt; aws ec2 detach-volume --volume-id $vid; let delay=2; while ! aws ec2 describe-volumes --volume-ids $vid | jq -re .Volumes[0].Attachments==[]; do if [[ $delay -gt 30 ]]; then aws ec2 detach-volume --force --volume-id $vid; break; fi; sleep $delay; let delay=$delay*2; done; aws ec2 delete-volume --volume-id $vid" EXIT
+echo Setting up SIGEXIT handler >&2
+trap "umount /mnt || umount -l /mnt; aws ec2 detach-volume --volume-id $vid; let delay=10; while ! aws ec2 describe-volumes --volume-ids $vid | jq -re .Volumes[0].Attachments==[]; do if [[ \$delay -gt 80 ]]; then aws ec2 detach-volume --force --volume-id $vid; sleep 120; break; fi; sleep \$delay; let delay=\$delay*2; done; aws ec2 delete-volume --volume-id $vid" EXIT
```

Comment:
We had to add `sleep 120` after `detach-volume --force` to give it a chance to work.   Otherwise `delete-volume` fails.